core/system_impl.cpp

```diff
@@ -611,47 +610,48 @@ MAVLinkCommands::Result SystemImpl::make_command_flight_mode(FlightMode flight_m
             break;
         default:
             LogErr() << "Unknown Flight mode.";
-            return MAVLinkCommands::Result::UNKNOWN_ERROR;
+            MAVLinkCommands::CommandLong empty_command{};
+            return std::make_pair<>(MAVLinkCommands::Result::UNKNOWN_ERROR, empty_command);
     }
 
-    command = MAVLinkCommands::CommandLong(); // wipe off caller's data, if any.
+    MAVLinkCommands::CommandLong command{};
 
     command.command = MAV_CMD_DO_SET_MODE;
     command.params.param1 = float(mode);
     command.params.param2 = float(custom_mode);
     command.params.param3 = float(custom_sub_mode);
     command.target_component_id = component_id;
 
-    return MAVLinkCommands::Result::SUCCESS;
+    return std::make_pair<>(MAVLinkCommands::Result::SUCCESS, command);
 }
 
 MAVLinkCommands::Result SystemImpl::set_flight_mode(FlightMode system_mode, uint8_t component_id)
 {
-    MAVLinkCommands::CommandLong command{};
+    std::pair<MAVLinkCommands::Result, MAVLinkCommands::CommandLong> result =
```

Comment:
You're right, it would be simpler, however, then the reader does not know what `first` and `second` refer to. That was my thought when writing it out.

Reply:
I read more about it here: https://stackoverflow.com/questions/6900459/the-new-keyword-auto-when-should-it-be-used-to-declare-a-variable-type
And the bottom line is:
> So the bottomline is: the keyword auto might increase or decrease clarity and readability of your code, depending on the context. If the context makes it clear what type it is, or at least how it should be used (in case of standard container iterator) or the knowledge of the actual type is not even needed (such as in expression templates), then auto should be used, and if the context doesn't make it clear and isn't very common (such as the second case above), then it should better be avoided.

According to that the type is not clear from context or common, so `auto` should rather not be used.

Reply:
Oh! Yes. It makes sense. Thanks Julian for the justification!

Reply:
Cool, so we agree :smiley:

Reply:
Yes :smiley: